modules/tinymce/src/plugins/fullscreen/test/ts/browser/FullScreenPluginTest.ts

```diff
@@ -1,72 +1,134 @@
-import { Assertions, Chain, Guard, Log, Pipeline, UiFinder } from '@ephox/agar';
+import { Assertions, Chain, Log, NamedChain, Pipeline, UiFinder } from '@ephox/agar';
 import { UnitTest } from '@ephox/bedrock-client';
-import { Cell } from '@ephox/katamari';
-import { Editor as McEditor } from '@ephox/mcagar';
-import { Attribute, Html, SelectorFind, SugarBody, SugarElement } from '@ephox/sugar';
+import { Arr, Cell } from '@ephox/katamari';
+import { ApiChains, Editor as McEditor } from '@ephox/mcagar';
+import { Attribute, Classes, Css, Insert, Remove, Html, SelectorFind, SugarBody, SugarElement, SugarShadowDom } from '@ephox/sugar';
 import Editor from 'tinymce/core/api/Editor';
 
 import FullscreenPlugin from 'tinymce/plugins/fullscreen/Plugin';
 import LinkPlugin from 'tinymce/plugins/link/Plugin';
 import SilverTheme from 'tinymce/themes/silver/Theme';
 
-export const cWaitForDialog = (ariaLabel) =>
-  Chain.control(
-    Chain.fromChainsWith(SugarBody.body(), [
-      UiFinder.cWaitFor('Waiting for dialog', '[role="dialog"]'),
-      Chain.op((dialog) => {
-        if (Attribute.has(dialog, 'aria-labelledby')) {
-          const labelledby = Attribute.get(dialog, 'aria-labelledby');
-          const dialogLabel = SelectorFind.descendant<HTMLLabelElement>(dialog, '#' + labelledby).getOrDie('Could not find labelledby');
-          Assertions.assertEq('Checking label text', ariaLabel, Html.get(dialogLabel));
-        } else {
-          throw new Error('Dialog did not have an aria-labelledby');
+interface Config {
+  label: string;
+  setup: () => Chain<Record<string, any>, Record<string, any>>;
+  cleanup: () => Chain<any, any>;
+}
+
+const getContentContainer = (editor: Editor) =>
+  SugarShadowDom.getContentContainer(SugarShadowDom.getRootNode(SugarElement.fromDom(editor.getElement())));
+
+const cCloseOnlyWindow = Chain.label(
+  'Close window',
+  Chain.fromChains([
+    NamedChain.read('editor',
+      Chain.op((editor: Editor) => {
+        const dialogs = () => UiFinder.findAllIn(getContentContainer(editor), '[role="dialog"]');
+        Assertions.assertEq('One window exists', 1, dialogs().length);
+        editor.windowManager.close();
+        Assertions.assertEq('No windows exist', 0, dialogs().length);
+      })
+    )
+  ])
+);
+
+const cWaitForDialog = (ariaLabel: string) =>
+  Chain.label(
+    'Looking for dialog with an aria-label: ' + ariaLabel,
+    Chain.fromChains([
+      NamedChain.direct('editor', Chain.mapper(getContentContainer), 'contentContainer'),
```

Comment:
This isn't a place named chains should be used as the items aren't used out of order or multiple times. Also where is `editor` coming from as that's never assigned and there's no `NamedChain.asChain` call? I suspect you're attempting to use something that just happens to work but shouldn't.

Anyways, this could be:
```
Chain.mapper(getContentContainer),
UiFinder.cWaitFor('Waiting for dialog', '[role="dialog"]'),
Chain.op((dialog) => { ... }
```

Reply:
To clarify more, you're mixing NamedChains with Chains. You're also trying to pass the named chain state across multiple functions, which makes it **very** hard to follow and messy, as you never know what state you have.